### PR TITLE
Planes & Layers part 2: Plane Masters

### PR DESCRIPTION
**Pull request by Generalcamo**
Part 2 of the PR series to bring /tg/'s and bay's plane masters to Aurora, the lack of which is blocking several features we want.

This ports over the easier to understand Bay version of plane masters, which is detailed in the relevant readme file in the code. Example effect code for a warp effect is also in, which has been implemented for gravity catapults.

Relies on #18741

### CODE REVIEW

**Comment by Generalcamo** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

File: code/__DEFINES/__initialization.dm [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#define INITIALIZE_IMMEDIATE(X) ##X/New(loc, ...){\
+	..();\
+	if(!(flags_1 & INITIALIZED_1)) {\
+		var/previous_initialized_value = SSatoms.initialized;\
+		SSatoms.initialized = INITIALIZATION_INNEW_MAPLOAD;\
+		args[1] = TRUE;\
+		SSatoms.InitAtom(src, FALSE, args);\
+		SSatoms.initialized = previous_initialized_value;\
+	}\
+}
```
Comment (FluffyGhoster) [inline code/__DEFINES/__initialization.dm:33]: Leave them in their file where they were, which is aligned with TG's version

File: code/__DEFINES/_macros.dm [hunk 2]
```diff
@@ -182,3 +182,20 @@
...
+
+//Bay uses the check before deleting something, to ensure it's not null, TG does not, and since I don't want to go hunt down 3 billion runtimes, we keep bay's version
+
+///Delete and null the reference, if the object isn't null
+#define QDEL_NULL(item) if(item) { qdel(item) ; item = null }
+
+#define QDEL_LIST(L) if(L) { for(var/I in L) qdel(I); L.Cut(); }
+#define QDEL_LIST_IN(L, time) addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(______qdel_list_wrapper), L), time, TIMER_STOPPABLE)
+#define QDEL_LIST_ASSOC(L) if(L) { for(var/I in L) { qdel(L[I]); qdel(I); } L.Cut(); }
+#define QDEL_LIST_ASSOC_VAL(L) if(L) { for(var/I in L) qdel(L[I]); L.Cut(); }
```
Comment (FluffyGhoster) [inline code/__DEFINES/_macros.dm:201]: Same, leave them where they were

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

**Comment by FluffyGhoster** [comment]
It looks like this/the previous PR is/are so far causing these issues I noticed:

- Space is white, lacking the icon
- Character preview is black, without the character nor background
- Windows are rendered above shutters
- There's a newscaster in the middle of the central hall, D2 for some reason
- There's also a part of it in D3, or at least it's visible through the floor
- Air alarm in D3 too, on the other side
- Zmimic does not seem to shade the lower layers properly
- People seem to be masked below floorlights
- Skrells seems to go below the railings in D2, for some reason; it doesn't seem to do that with humans - Same with the bluespace beacon outside security in D2
- Sensors and starmap seems to have issues too, though I was only told so, so I'm not sure if it actually is true

**Comment by RustingWithYou** [comment]
> Sensors and starmap seems to have issues too, though I was only told so, so I'm not sure if it actually is true

Can confirm - overmap effects are layered below the overmap space turfs for some reason, meaning that navigation is effectively impossible.

This has also caused some bugs with mapping - pipes are now layered under all turfs in StrongDMM, making atmos mapping very difficult

**Review by DreamySkrell — APPROVED** [review]

File: html/changelogs/GeneralCamo - Plane Masters.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - backend: "Ported plane masters from Bay."
+  - backend: "Ported some Z-Mimic changes/improvements/bugfixes from Nebula/Bay."
+  - bugfix: "Floor lights no longer render under floor decals."
```
Comment (NonQueueingMatt) [inline html/changelogs/GeneralCamo - Plane Masters.yml:43]:
```suggestion
  - refactor: "Ported plane masters from Bay."
  - refactor: "Ported some Z-Mimic changes/improvements/bugfixes from Nebula/Bay."
  - bugfix: "Floor lights no longer render under floor decals."
```

**Pull request merged**